### PR TITLE
Added Listener for Configuration Changes

### PR DESCRIPTION
**Pull request by vrenjith**
Changes proposed in this pull request:
- Added an `ItemListener` for listening for configuration changes

- [n] Includes tests covering the new functionality?
- [y] Ready for review
- [y] Follows CONTRIBUTING rules

@nickgrealy

### CODE REVIEW

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+    }
+
+    @Override
+    public void onCreated(final Item item)
+    {
+        this.getParentPluginDescriptor().processEvent(Event.ITEM_CREATED, log, new HashMap<Object, Object>() {{
+            put("item", item);
+        }});
+        super.onCreated(item);
```
Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: Are you sure that this is needed? I'm expecting that same event would be handled by different `ItemListener` instances anyway.

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        if (parentPluginDescriptorOverride != null){
+            return parentPluginDescriptorOverride;
+        } else {
+            return Jenkins.getInstance().getPlugin(GlobalEventsPlugin.class).getDescriptor();
+        }
+    }
+
+    @Override
+    public void onCreated(final Item item)
+    {
```
Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: Please, move `{` to previous line.

Comment (vrenjith) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: Updated

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java (hunk 1 above)
Comment (vrenjith) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: In my tests, the log message was only printed once came only once.

Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: According source code of `ItemListener.onCreated`:
```
public void onCreated(Item item) {
    }
```
It doesn't do anything. Why do we need to call it in this case?

**Comment by nickgrealy** [comment]
@Jimilian - are you happy to approve the changes?

**Comment by Jimilian** [comment]
@nickgrealy To be honest, nope. I hope that @vrenjith can say why do we need to call parent method. Even in case then this method is not doing anything (empty).

I very appreciate any PR, so, we can merge it and after that I will remove code that I don't like. But maybe @vrenjith knows something about these methods that I don't.

**Comment by nickgrealy** [comment]
@Jimilian - no problem, I'll wait until those changes have been discussed (tests would be good!) and let you perform the merge. 👍

**Comment by vrenjith** [comment]
I haven't checked the implementation of `ItemListener` and to be on the safe side I assumed it is always better to call the parent. Hence retained the `super` calls. But if @Jimilian thinks that it is unnecessary, I can just remove them and update the PR

**Comment by Jimilian** [comment]
@vrenjith Yes, it would be perfect. As you can see in any other listener in this plugin (i.e., https://github.com/jenkinsci/groovy-events-listener-plugin/blob/master/src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalRunListener.java) we don't call parent function. Otherwise it would be broken [Observer](https://en.wikipedia.org/wiki/Observer_pattern) pattern.

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+/**
+ * Created by Renjith Pillai (i306570) on 15/09/16.
+ */
+@Extension
+public class GlobalItemListener extends ItemListener
+{
```
Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: This bracket should be on previous line as well.

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.listeners.ItemListener;
+import jenkins.model.Jenkins;
+
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+/**
+ * Created by Renjith Pillai (i306570) on 15/09/16.
```
Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalItemListener.java]: And I'm not sure that we are keeping information about creator/ownership in source code itself. Git shows it much better.

**Comment by Jimilian** [comment]
@vrenjith I uncommented checking java version and merged you PR. Thanks a lot for passion and patience:)

**Comment by Jimilian** [comment]
@nickgrealy, could you release new version, please?
We found a beautiful use case there this feature could rock :)

**Comment by nickgrealy** [comment]
@Jimilian yep sure! Give me 15 mins...

**Comment by nickgrealy** [comment]
@Jimilian - just checking, are you happy with the state of the current master branch?

There's been a lot of changes since the last release -  dc5f51d

Including this line - https://github.com/jenkinsci/groovy-events-listener-plugin/blob/master/build.gradle#L82 - which re-introduces your issue - https://github.com/jenkinsci/groovy-events-listener-plugin/issues/21

**Comment by Jimilian** [comment]
@nickgrealy Oups, I forgot about https://github.com/jenkinsci/groovy-events-listener-plugin/blob/master/build.gradle#L82.

**Pull request merged**